### PR TITLE
Digraph implementation - yeeting 2.6.0pre2 into base.

### PR DESCRIPTION
**Pull request by Lyxal**

### CODE REVIEW

**Review by pxeger — APPROVED** [review]
Lots of comments, but \almost all minor things that shouldn't prevent merging, so I'm approving

File: vyxal/elements.py
Comment (cgccuser) [reply vyxal/elements.py:4180]: Because of scoping. Simply doing `stack = [stack]` was making the tests fail because a new local stack was being created. It appears to be a necessary evil as long as we make `stack` a global variable like that.

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:1441]: Do you mean `vectorise: false`? I agree, it shouldn't be vectorized, the last case should be `1` (and `[]` should probably be `0` because of Python)

File: documents/knowledge/elements.yaml
Comment (cgccuser) [inline documents/knowledge/elements.yaml:1578]: By the way, do we want to preserve the original behavior here?

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:1626]: wdym?

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:1667]: Nope, `str-num` takes precedence because `'abc' * 3` is `'abcabcabc'`, but if we treated it like `any-num`, then it would be `['abc', 'abc', 'abc']`. I'm not convinced the str-num overload is needed at all, since we have `*` anyway, but /shrug.

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:1942]: Nope. The implementation currently seems to guarantee that, but we need a better way of generating tests that check equality of unordered lists.

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:2409]: I'm sure lyxal's hand accidentally slipped from one side of the keyboard to another :P

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:2441]: Methinks it is.

File: vyxal/elements.py
Comment (pxeger) [reply vyxal/elements.py:4180]: Ah ok, that makes sense. What I was mainly commenting on was actually the `deep_copy` - I understand it needs to be copied once (otherwise the stack would contain a circular reference to itself), but why a deep copy?.

By the way, this can be, more idiomatically:

```
stack[:] = [stack.copy()]
# or deep_copy(stack)
```

File: documents/knowledge/elements.yaml
Comment (pxeger) [reply documents/knowledge/elements.yaml:1626]: ok, fair enough; I was just wondering why you'd swap them because it doesn't matter what order you put keyword arguments in

File: documents/knowledge/elements.yaml
Comment (cgccuser) [inline documents/knowledge/elements.yaml:3309]: I don't understand why we can't merge this and ∆q into one for a truly general quadratic solver.

File: documents/knowledge/elements.yaml
Comment (pxeger) [reply documents/knowledge/elements.yaml:1667]: Ah, ok - I thought "repeat a b times" meant `a * b` as well, but it's actually `[a] * b`. In that case, this string behaviour is now annoyingly inconsistent with lists - can I suggest we don't change this, given we have `*` already?

File: documents/knowledge/elements.yaml
Comment (cgccuser) [reply documents/knowledge/elements.yaml:3558]: Would be a little hard to implement, but probably worth it.

File: documents/knowledge/elements.yaml
Comment (pxeger) [reply documents/knowledge/elements.yaml:2409]: It wasn't Lyxal, it was @razetime lol

https://github.com/Vyxal/Vyxal/blame/d5c0aef71cfffd2e47d3f0f60103e9a100ad7d91/documents/knowledge/elements.yaml#L2149

File: documents/knowledge/yaml_to_txt.py
Comment (cgccuser) [inline documents/knowledge/yaml_to_txt.py:1]: Do we really need a txt?

File: files/addition.vy
Comment (cgccuser) [reply files/addition.vy:1]: Not golfy

File: vyxal/helpers.py
Comment (cgccuser) [reply vyxal/helpers.py:80]: Why `self`? It isn't inside `LazyList`, right?

File: vyxal/helpers.py
Comment (cgccuser) [reply vyxal/helpers.py:86]: I think it should be `vyxalify`?

File: vyxal/helpers.py
Comment (cgccuser) [inline vyxal/helpers.py:315]: Make this docstring a single line?

**Review by cgccuser — APPROVED** [review]
There are things to be fixed, but I think we can merge now and fix those in a separate (short-lived) branch.

File: documents/knowledge/elements.yaml
Comment (pxeger) [reply documents/knowledge/elements.yaml:3309]: I think the point is this is a shortcut for if you know that `a` is definitely going to be 1, and `∆q` is for if you know `c` is definitely `0`, so it's a little bit shorter.

File: vyxal/elements.py
Comment (cgccuser) [reply vyxal/elements.py:4180]: No real reason for a deep copy, honestly, I was just being unpythonic.

File: vyxal/helpers.py
Comment (pxeger) [reply vyxal/helpers.py:80]: You're right, GitHub's margins are just too wide so it looked like it was indented

File: templates/main.js
Comment (chunkybanana) [reply templates/main.js:18]: Yeah, that one's Aaron's.

File: vyxal/helpers.py
Comment (chunkybanana) [reply vyxal/helpers.py:139]: Agreed

File: vyxal/helpers.py
Comment (chunkybanana) [reply vyxal/helpers.py:250]: I... did not knoow about that.

**Review by chunkybanana — APPROVED** [review]
looks pretty poggers, the minor things can wait

File: documents/knowledge/elements.yaml
Comment (Lyxal) [reply documents/knowledge/elements.yaml:2317]: I needed a test case where the second string had an even number of characters leave me alone :p

File: .github/workflows/linter.yaml
Comment (Lyxal) [reply .github/workflows/linter.yaml:1]: If I didn't modify it, it wouldn't have passed the linter check. But sure, it could have been added after merging this branch and that probably would have been better.

File: documents/knowledge/elements.yaml
Comment (Lyxal) [reply documents/knowledge/elements.yaml:3501]: Erm, `•` already does that, and has been doing so for a while.

File: documents/knowledge/elements.yaml
Comment (Lyxal) [reply documents/knowledge/elements.yaml:3506]: BTW that wasn't me that was Codex that named it.

File: scripts/vyxal
Comment (Lyxal) [reply scripts/vyxal:1]: I just copied what jelly does, so I don't exactly know if it works in this context.

File: vyxal/elements.py
Comment (Lyxal) [reply vyxal/elements.py:38]: Correct. There's a few edge cases with Sympy being too precise for float comparison when using the default value provided by sys. (trust me I tried using it and it was too small)

File: vyxal/helpers.py
Comment (Lyxal) [reply vyxal/helpers.py:86]: This is because it's for converting Sympy solutions for an equation (which are `{symbol: value}` pairs) to a list of lists.

The str(key) is so that the key isn't left as a symbol.

File: documents/knowledge/yaml_to_txt.py
Comment (razetime) [reply documents/knowledge/yaml_to_txt.py:1]: i think it's good to have an autogenerated txt for documentation. we can probably gitignore the generated txt anyway

File: documents/knowledge/elements.yaml
Comment (Lyxal) [reply documents/knowledge/elements.yaml:1441]: This was covered in #227. However the docs do need updating

File: documents/knowledge/elements.yaml
Comment (Lyxal) [reply documents/knowledge/elements.yaml:768]: The only problem with this is that if you make `K` prefixes for strings, `¦` becomes useless on strings (because it's cumulative sums). But then if you make `K` contiguous sublists for lists, there's no one-byte way to get list prefixes.

File: vyxal/elements.py
Comment (Lyxal) [reply vyxal/elements.py:1005]: because wrapping element functions in `lazylist` doesn't seem to work - 

```
NameError: name 'wrapped' is not defined
```

**Pull request merged**
